pkg/operation/seed/seed.go

```diff
@@ -808,51 +804,39 @@ func RunReconcileSeedFlow(
 					LoadBalancerIP:  handler.SNI.Ingress.ServiceExternalIP,
 					Labels:          gutil.GetMandatoryExposureClassHandlerSNILabels(handler.SNI.Ingress.Labels, handler.Name),
 				},
-				*handler.SNI.Ingress.Namespace,
-				chartApplier,
-				charts.Path,
-				seedClient,
-			))
-		}
+				Namespace: *handler.SNI.Ingress.Namespace,
+				ChartPath: filepath.Join(charts.Path, "istio", "istio-ingress"),
+			})
 
-		if err := component.OpWaiter(istioDeployers...).Deploy(ctx); err != nil {
-			return err
+			istioProxyGateway = append(istioProxyGateway, &istio.IstioProxyProtocol{
+				Values: &istio.ProxyValues{
+					Labels: gutil.GetMandatoryExposureClassHandlerSNILabels(handler.SNI.Ingress.Labels, handler.Name),
+				},
+				Namespace: *handler.SNI.Ingress.Namespace,
+				ChartPath: filepath.Join(charts.Path, "istio", "istio-proxy-protocol"),
+			})
 		}
-	}
 
-	var proxyGatewayDeployers = []component.DeployWaiter{
-		istio.NewProxyProtocolGateway(
-			&istio.ProxyValues{
-				Labels: conf.SNI.Ingress.Labels,
-			},
-			*conf.SNI.Ingress.Namespace,
-			chartApplier,
-			charts.Path,
-		),
-	}
+		if !gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) {
+			istioProxyGateway = nil
+		}
 
-	for _, handler := range conf.ExposureClassHandlers {
-		proxyGatewayDeployers = append(proxyGatewayDeployers, istio.NewProxyProtocolGateway(
-			&istio.ProxyValues{
-				Labels: gutil.GetMandatoryExposureClassHandlerSNILabels(handler.SNI.Ingress.Labels, handler.Name),
+		istiod := istio.NewIstio(
+			seedClient,
+			chartRenderer,
+			&istio.IstiodValues{
+				TrustDomain: gardencorev1beta1.DefaultDomain,
+				Image:       istiodImage.String(),
 			},
-			*handler.SNI.Ingress.Namespace,
-			chartApplier,
+			common.IstioNamespace,
 			charts.Path,
-		))
-	}
+			istioIngressGateway,
+			istioProxyGateway,
+		)
+		istioDeployers := []component.DeployWaiter{istioCRDs, istiod}
 
-	if gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) {
-		for _, proxyDeployer := range proxyGatewayDeployers {
-			if err := proxyDeployer.Deploy(ctx); err != nil {
-				return err
-			}
-		}
-	} else {
-		for _, proxyDeployer := range proxyGatewayDeployers {
-			if err := proxyDeployer.Destroy(ctx); err != nil {
-				return err
-			}
+		if err := component.OpWaiter(istioDeployers...).Deploy(ctx); err != nil {
```

Comment:
Can this rather be moved to `components.go` and to https://github.com/gardener/gardener/blob/master/pkg/operation/seed/seed.go#L964?

Reply:
We should not do it, cause we probably don't want to deploy seed-bootstrap charts before istio which is less important than istio.

Reply:
Why would this be an issue? I am not aware that we require a certain deployment order in the seed controller.

Reply:
I though we should give priority to istio over fluent and other seed bootstrap charts. I will adapt the required changes.